Lib/imaplib.py

```diff
@@ -318,9 +318,11 @@ def shutdown(self):
         self.file.close()
         try:
             self.sock.shutdown(socket.SHUT_RDWR)
-        except OSError as e:
-            # The server might already have closed the connection
-            if e.errno != errno.ENOTCONN:
+        except OSError as exc:
+            # The server might already have closed the connection.
+            # Windows error 10022: An invalid operation was attempted.
```

Comment:
> (In my experience invalid operation is indeed what windows raises when you attempt an operation on a closed socket).

I failed to reproduce the error on Windows on a closed socket. If the socket is closed, its socket handle becomes -1. Calling shutdown() with the socket handle -1 is not a good idea. It would be better to raise an exception rather than calling shutdown() if the socket is known to be closed, but I'm too shy to propose such change.